std-bits/table/src/main/java/org/enso/table/read/QuoteHelper.java

```diff
@@ -0,0 +1,54 @@
+package org.enso.table.read;
+
+import java.util.function.Consumer;
+
+/**
+ * A helper class providing functionality of parsing optionally quoted strings and detecting
+ * mismatched quotes.
+ *
+ * <p>The tool is meant for use with the output of the {@code CsvParser} with the setting {@code
+ * keepQuotes} set to true. The strings which were quoted will keep the enclosing quotes, but any
+ * escaped quotes within the string are correctly parsed (for example, if the escaped quote is `\"`,
+ * it will appear as already parsed `"` quote inside of the string, so no escape processing is
+ * performed at this stage, because it was done earlier by the parser.
+ */
+public class QuoteHelper {
+  private final Consumer<String> mismatchedQuoteCallback;
+  private final char quoteCharacter;
+
+  public QuoteHelper(Consumer<String> mismatchedQuoteCallback, char quoteCharacter) {
+    this.mismatchedQuoteCallback = mismatchedQuoteCallback;
+    this.quoteCharacter = quoteCharacter;
+  }
+
+  /**
+   * Strips enclosing quotes from the string, if they are present.
+   *
+   * <p>If the quote appears only at one end of the string, a mismatched quote callback is called to
+   * report the issue.
+   */
+  public String stripQuotes(String text) {
+    int last = text.length() - 1;
+    if (text.isEmpty()
+        || (text.charAt(0) != quoteCharacter && text.charAt(last) != quoteCharacter)) {
```

Comment:
```java
    int last;
    if (text.isEmpty()
        || (text.charAt(0) != quoteCharacter && text.charAt(last = text.length() - 1) != quoteCharacter)) {
```

Tiny optimisation

Reply:
In my opinion this is much much less readable. And it's such a micro-optimization, I don't think it would affect performance at all, knowing how the operations are usually pipelined anyway.

Reply:
After a discussion I understand that the reason for this was what if `length()` was a costly (O(N)?) operation. But at least for `String` that we use here, it is very fast (just a field read), so I don't think this optimization is worth it.